Lombiq.Hosting.Tenants.Maintenance/Readme.md

```diff
@@ -196,3 +196,41 @@ The following configuration options are available to set the enabled features:
   }
 }
 ```
+
+### `Lombiq.Hosting.Tenants.Maintenance.StaggeredTenantWakeUp`
+
+Adds a page to the admin under Multi-Tenancy/Staggered Tenant Wake Up that allows you to start all tenants shells in a staggered way. Waking up all the tenants this way ensures that migrations (and maintenance, should you have them) are all executed without having to start each tenant manually and also avoids saturating hardware resources (when configured correctly), such as the database access.
+
+You can edit the maintenance options directly on the Multi-Tenancy / Staggered Tenant Wake Up page if you click on the edit button.
+
+- Batch Size: The number of tenants processed in one iteration.
+- Time Span Between Batches: The amount of time to wait between processing batches.
+- Run In Parallel: Determines whether the number of tenants defined by the batch size are woken up in parallel or sequentially.
+- Current Version: The current version of the staggered tenant wake-up process. Usually this should not be set manually.
+
+Functions on the page:
+
+- Start new: Starts the staggered tenant wake-up process with a new version number.
+- Continue: Continues most recent wake-up process if it was paused.
+- Pause: Pauses the currently running wake-up process.
+- Reset current version: Runs the staggered tenant wake-up process again with the current version.
+- Edit: Allows you to edit the options listed above. You can only save if there is no maintenance running.
+
+You can initialize the configuration with the following app settings:
+
+```json
+{
+  "OrchardCore": {
+    "Lombiq_Hosting_Tenants_Maintenance": {
+      "StaggeredTenantWakeUp": {
+        "BatchIntervalSeconds": 5,
+        "BatchSize": 5,
+        "RunParallel": false,
+        "RunOnStartup": true
+      }
+    }
+  }
+}
+```
+
+- RunOnStartup: When set to true, the staggered tenant wake-up process will start a new version if a new deployment was done, or it will continue the current wake-up process if it hasn't finished yet. This is useful if you want to run the staggered tenant wake-up process automatically when the application starts. If you set this to false, you will have to start it manually from the admin page.
```

Comment:
Based on the text (haven't tested yet): If a new deployment happens, the process should start from scratch, otherwise some tenants might not run the changes that were just deployed.

Reply:
Yes!